app/controllers/projects_controller.rb

```diff
@@ -0,0 +1,79 @@
+class ProjectsController < ApplicationController
+  before_action :set_project, only: [:show, :edit, :update, :destroy]
+  before_action :validates_role, only: [:new, :edit, :update, :create]
+  before_action :validates_admin, only: :destroy
+
+  def index
+    @projects = Project.all
+  end
+
+  def show
+  end
+
+  def new
+    @project = Project.new
+    get_clients
+    get_users
+  end
+
+  def edit
+    get_clients
+    get_users
+  end
+
+  def create
+    @project = Project.new(project_params)
+    params[:users][:id].each do |user|
+      if !user.empty?
+        @project.assignments.build(user_id: user)
+      end
+    end
+
+    if @project.save
+      redirect_to @project, notice: 'Project was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @project.update(project_params)
```

Comment:
even if we do go with your logic of adding users in create..there is no code in update to support that.